sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -434,17 +423,92 @@ export class ServiceBusManagementClient extends ServiceClient {
    * @throws `RestError` with code that is a value from the standard set of HTTP status codes as documented at
    * https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=netframework-4.8
    */
-  async getQueues(options?: ListRequestOptions & OperationOptions): Promise<QueuesResponse> {
+  private async listQueues(
+    options?: ListRequestOptions & OperationOptions
+  ): Promise<EntitiesResponse<QueueDescription>> {
     log.httpAtomXml(`Performing management operation - listQueues() with options: ${options}`);
     const response: HttpOperationResponse = await this.listResources(
       "$Resources/Queues",
       options,
       this.queueResourceSerializer
     );
-
     return this.buildListQueuesResponse(response);
   }
 
+  private async *listQueuesPage(
+    marker?: number,
+    options: OperationOptions & Pick<PageSettings, "maxPageSize"> = {}
+  ): AsyncIterableIterator<EntitiesResponse<QueueDescription>> {
+    let listResponse;
+    if (!!marker || marker === undefined) {
```

Comment:
so this would only be false when `marker` is 0? Is it simpler to write `marker !== 0`?

Reply:
Thanks for asking, I wanted to revisit this.
This was the logic when the marker is a "string" to not allow empty strings I guess, for example: storage.

`marker === undefined || marker >= 0` seems more appropriate for service-bus since the marker is a number and negative values are not allowed. 
"undefined" or "0" are allowed to start iterating from the beginning, the marker then becomes a positive integer if there are entities to list for, and becomes "undefined" at the end which also ends the `do.. while` loop.

I'm going with `marker === undefined || marker >= 0`  and will add some tests.

Reply:
Updated and added tests